datafusion/core/tests/sql/select.rs

```diff
@@ -176,27 +175,6 @@ async fn prepared_statement_type_coercion() -> Result<()> {
     Ok(())
 }
 
-#[tokio::test]
-async fn prepared_statement_invalid_types() -> Result<()> {
```

Comment:
Remove this test because now parameters will be cast to the target type. 
This is also the behavior of PostgreSQL.
```sh
psql=> prepare p(int) as select 100+$1;
PREPARE

psql=> execute p('100');
 ?column?
----------
      200
(1 row)

psql=> execute p(20.12);
 ?column?
----------
      120
(1 row)
```